src/inference/include/openvino/runtime/intel_myriad/hddl_properties.hpp

```diff
@@ -0,0 +1,157 @@
+//    RO properties
+/**
+ * @brief Metric to get a int of the device number, String value is METRIC_HDDL_DEVICE_NUM
+ */
+static constexpr Property<int, PropertyMutability::RO> deviceNum{"HDDL_DEVICE_NUM"};
+
+/**
+ * @brief Metric to get a std::vector<std::string> of device names, String value is METRIC_HDDL_DEVICE_NAME
+ */
+static constexpr Property<std::vector<std::string>, PropertyMutability::RO> deviceName{"HDDL_DEVICE_NAME"};
+
+/**
+ * @brief  Metric to get a std::vector<float> of device thermal, String value is METRIC_HDDL_DEVICE_THERMAL
+ */
+static constexpr Property<std::vector<float>, PropertyMutability::RO> deviceThermal{"HDDL_DEVICE_THERMAL"};
+
+/**
+ * @brief  Metric to get a std::vector<uint32> of device ids, String value is METRIC_HDDL_DEVICE_ID
+ */
+static constexpr Property<std::vector<unsigned int>, PropertyMutability::RO> deviceId{"HDDL_DEVICE_ID"};
+
+/**
+ * @brief  Metric to get a std::vector<int> of device subclasses, String value is METRIC_HDDL_DEVICE_SUBCLASS
+ */
+static constexpr Property<std::vector<int>, PropertyMutability::RO> deviceSubclass{"HDDL_DEVICE_SUBCLASS"};
+
+/**
+ * @brief  Metric to get a std::vector<uint32> of device total memory, String value is METRIC_HDDL_MEMORY_TOTAL
+ */
+static constexpr Property<std::vector<unsigned int>, PropertyMutability::RO> deviceMemoryTotal{
+    "HDDL_DEVICE_MEMORY_TOTAL"};
+
+/**
+ * @brief  Metric to get a std::vector<uint32> of device used memory, String value is METRIC_HDDL_DEVICE_MEMORY_USED
+ */
+static constexpr Property<std::vector<unsigned int>, PropertyMutability::RO> deviceMemoryUsed{
+    "HDDL_DEVICE_MEMORY_USED"};
+
+/**
+ * @brief  Metric to get a std::vector<float> of device utilization, String value is METRIC_HDDL_DEVICE_UTILIZATION
+ */
+static constexpr Property<std::vector<float>, PropertyMutability::RO> deviceUtilization{"HDDL_DEVICE_UTILIZATION"};
+
+/**
+ * @brief  Metric to get a std::vector<std::string> of stream ids, String value is METRIC_HDDL_DEVICE_STREAM_ID
+ */
+static constexpr Property<std::vector<std::string>, PropertyMutability::RO> streamId{"HDDL_STREAM_ID"};
+
+/**
+ * @brief  Metric to get a std::vector<std::string> of device tags, String value is METRIC_HDDL_DEVICE_TAG
+ */
+static constexpr Property<std::vector<std::string>, PropertyMutability::RO> deviceTag{"HDDL_DEVICE_TAG"};
+
+/**
+ * @brief  Metric to get a std::vector<int> of group ids, String value is METRIC_HDDL_GROUP_ID
+ */
+static constexpr Property<std::vector<int>, PropertyMutability::RO> groupId{"HDDL_GROUP_ID"};
+
+/**
+ * @brief  Metric to get a int number of device be using for group, String value is METRIC_HDDL_DEVICE_GROUP_USING_NUM
+ */
+static constexpr Property<int, PropertyMutability::RO> deviceGroupUsingNum{"HDDL_DEVICE_GROUP_USING_NUM"};
+
+/**
+ * @brief  Metric to get a int number of total device, String value is METRIC_HDDL_DEVICE_TOTAL_NUM
+ */
+static constexpr Property<int, PropertyMutability::RO> deviceTotalNum{"HDDL_DEVICE_TOTAL_NUM"};
```

Comment:
@ilyachur am I right that `device_total_num` naming should be used?

Reply:
I hope so

Reply:
Just to clarify:
Do I need to change all the property names from 'camelCase' to 'snake_case'?

Reply:
Renamed to snake_case